cmake/go.cmake

```diff
@@ -48,14 +48,25 @@ function(go_executable dest package)
     endif()
 
     set(depfile "${CMAKE_CURRENT_BINARY_DIR}/${dest}.d")
```

Comment:
NP: Since `depfile` is only needed when the CMake version is >= 3.7, we could move this initialization to line 61 below.